### PR TITLE
Roles

### PR DESCRIPTION
**Pull request by mrsanz**
sass not working to hide bullets for todo items but we're close enough...

to get started create an empty project, edit your _config.yaml to include this
```
remote_theme: https://github.com/mrsanz/jekyll-roles-theme@roles
```
then add a role file
```
---
layout: role
title: Scribe
time: Meeting Duration + 5m
description: The Scribe ensures that meeting notes are taken and shared.
---
```
if it matches the template it will show up.  right now only the scribe exists.

### CODE REVIEW

**Comment by rrrutledge** [comment]
> ```
> remote_theme: https://github.com/mrsanz/jekyll-roles-theme@roles
> ```

Do you have an example of this working?  I had to include

```
remote_theme: mrsanz/jekyll-roles-theme@roles
```

to get it to work.

**Comment by rrrutledge** [comment]
> time: Meeting Duration + 5m
description: The Scribe ensures that meeting notes are taken and shared.

Possible to have a default implementation of these fields?  I don't think that every consumer should have to type them out themselves as a generic entry should work fine.

**Comment by rrrutledge** [comment]
How does `_includes/default_roles/scribe` relate to `_roles/scribe.md`?  Is only 1 needed?

File: _includes/head.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ <!-- <base href="https://pages.github.nike.com/rrutle/pragmatic-programmer-community/" /> -->
```
Comment (rrrutledge) [inline _includes/head.html]: Remove Nike-specific reference.

File: _layouts/home.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        {{ content }}
+      </div>
+    </section>
+
+    <section class="section py-5">
+      <div class="container">
+        <h3 class="title">Roles</h3>
+        <div class="row">
+          {% for role in site.roles %}
+          {% unless site.data.config.excludes contains role.title %}
```
Comment (rrrutledge) [inline _layouts/home.html]: Recommend an include list instead of exclude.

**Comment by rrrutledge** [comment]
Does GHE support gem-based themes?

**Review by rrrutledge — COMMENTED** [review]
How do I set up the `chat` link?

**Comment by mrsanz** [comment]
for 

> remote_theme: https://github.com/mrsanz/jekyll-roles-theme@roles

I have this running but not deployed via pages.  will try that out.

File: _includes/head.html (hunk 1 above)
Comment (mrsanz) [reply _includes/head.html]: done

File: _layouts/home.html (hunk 2 above)
Comment (mrsanz) [reply _layouts/home.html]: removed the exception, now includes all roles.

**Comment by mrsanz** [comment]
> How do I set up the chat link?

Currently anything in the form (in the file home config `_config.yaml`)

``` 
links:
  chat: http://slack.com/foo/bar
  google: http://google.com
```
all turn into link references for each page

\[chat\]: http://slack.com/foo/bar
[google]: http://google.com

**Comment by mrsanz** [comment]
> Possible to have a default implementation of these fields? I don't think that every consumer should have to type them out themselves as a generic entry should work fine.

I thought of this, and unfortunately I don' think there is a simple way without making it completely static, e.g. its either built into the page and un-changeable or not. 

If they're opting into the role they're going to copy something from somewhere...  These things seem like the things you would customize for your meetups as well.  for instance the meeting time is known so for prgamatic programmer they could put 45 minutes in for the scribe role, vs the devcom meeting is longer.  I see this being a natural point of extension, with the master template living in the opt-in instructions.

**Comment by mrsanz** [comment]
> Does GHE support gem-based themes?

this is both a gem-based and a default theme.  I've noticed most robust themes including both options.  a Gem based theme just includes some metadata about the project.

**Comment by mrsanz** [comment]
@rrrutledge updated instructions in comment thread (moved config for links into the master config)  and fixed the broken links.

**Comment by mrsanz** [comment]
> I thought of this, and unfortunately I don' think there is a simple way without making it completely static, e.g. its either built into the page and un-changeable or not.

Ok I've taken a second look based on our discussion today @rrrutledge.  The homepage as well as the sub pages utilize this metadata (included from the role docs), there is not a way to add defaults without duplicating the inclusion of the default information in the templates.  Here' is the code that would be added to the role page if we wanted to add defaults to the role but then the home page for the roles wouldn't be populated other than titles.  Having these as data in each file really gives a lot of flexibility to these pages.

```
  {% capture title %}{{ page.title }}{% endcapture %}
  {% capture description %}{{ page.description }}{% endcapture %}
  {% capture time %}{{ page.time }}{% endcapture %}
  {% if title | downcase == "scribe" and description == nil %}
    {{ assign description = "default documenation" }}
  {% endif %}
  {% if title | downcase == "scribe" and time == nil %}
    {{ assign time = "default documentation" }}
  {% endif %}
```

**Comment by utanapishtim** [comment]
Satisfies the AC, LGTM

**Comment by rrrutledge** [comment]
Thanks for merging.  Excited to start using it!

**Pull request merged**